packages/vaex-core/vaex/dataframe.py

```diff
@@ -7036,36 +7036,50 @@ def export_json(self, to, progress=None, chunk_size=default_chunk_size, parallel
         :param parallel: {evaluate_parallel}
         :param fs_options: {fs_options}
         :param fs: {fs}
+        :param backend: Which backend to use for writting the JSON file. Can be "pandas" or "json".
+        :param lines: If True, each row is written as a separate JSON record. If False, dataframe is written as a list of JSON records.
         :return:
         """
+
         json = None  # we may want to pass the module as parameter to use a faster library
         import json as json_std
         json = json or json_std
 
-        # not sure if we want to use pandas, it will treat datetime for us, but will convert null to nan
-        use_pandas = True
-
         # we take on the '[' and ']' from each chunk, and insert it back ourselves
         # and we also need to but ',' between each chunk
         with vaex.progress.tree(progress, title="export(json)"), vaex.file.open(path=to, mode='wb', fs_options=fs_options, fs=fs) as f:
-            f.write(b"[")
+            if not lines:
+                f.write(b"[")
             first = True
-            if use_pandas:
+            if backend == 'pandas':
                 for _i1, _i2, df in self.to_pandas_df(chunk_size=chunk_size, parallel=parallel):
                     if not first:
-                        f.write(b", ")
+                        if not lines:
+                            f.write(b", ")
                     first = False
                     f_temp = io.StringIO()
-                    df.to_json(f_temp, orient='records')
-                    f.write(f_temp.getvalue()[1:-1].encode('utf8'))
-            else:
+                    df.to_json(f_temp, orient='records', lines=lines)
+                    if lines:
+                        f.write(f_temp.getvalue().encode('utf8'))
+                    else:
+                        f.write(f_temp.getvalue()[1:-1].encode('utf8'))
+            elif backend == 'json':
                 for _i1, _i2, records in self.to_records(chunk_size=chunk_size, parallel=parallel):
                     if not first:
-                        f.write(b", ")
+                        if not lines:
+                            f.write(b", ")
                     first = False
                     raw = json.dumps(records)[1:-1]
+                    if lines:
+                        if not first:
+                            raw = raw.replace('},', '}\n')
```

Comment:
if we want the `from_json_arrow` to work, i can't get it to work otherwise.. it requires that each json record is on a new line, and cant end with a trailing comma. I do understand your concerns tho..